molr-mole-core/src/main/java/io/molr/mole/core/support/MissionPredicates.java

```diff
@@ -0,0 +1,64 @@
+package io.molr.mole.core.support;
+
+import io.molr.commons.domain.RunState;
+
+import java.util.function.Predicate;
+
+/**
+ * A general purpose validator for objects.
+ *
+ * @param <T> the type for object being validated
+ * @author HimanshuSahu31
+ */
+public class MissionPredicates<T> {
+    private Predicate<T> validator;
+
+    /**
+     * @param object an {@link Object} that represents the initial value
+     */
+    private MissionPredicates(T object) {
+        validator = o -> o.equals(object);
+    }
+
+    /**
+     * Factory method to get validator for {@link RunState}
+     *
+     * @param runState the initial value {@link RunState} for predicate function
+     * @return a {@link MissionPredicates} for validating {@link RunState}
+     */
+    public static MissionPredicates<RunState> runStatePredicate(RunState runState) {
```

Comment:
I think the methods in this class could return real instances of Predicate. ... And if I look at the usage, I think it should be `Predicate<MissionState>`, not of RunState. ... Then, also the `and()` method would make sense ...

Further, as this class has a plural name, it suggests to contain only static methods and it should never be instantiated.

Further, as it is implemented now, the T would always be RunState, so one would not need the type parameter T.... 

I could imagine something like:

```
public static Predicate<MissionState> runStateEquals(RunState runState) {
...
}

public static Predicate<MissionState> curserIsAt(Block block) {
.... // no need to implement this now, just as a concept ...
}

....
```